### DISCUSSION TITLE
Need some help with Get-FalconUninstallToken

### DISCUSSION THREAD

**Discussion by dgendera** [CrowdStrike/psfalcon#300] [Q&A]
This might not be correct forum to ask this question as it's more PowerShell related but I'm going to try anyway and hopefully somebody can provide some help with this. 

Checking the code I see that Get-FalconUninstallToken has as paramter "-id" and the oauth2.ps1 needs clientid and clientsecret

When I run Get-FalconUninstallToken -Id 1
it prompts me for ClientID, ClientSecret

is there some way I can pass these as part of the command line I've tried various combinations but not succeeding.   We would like to have this script run silently to un-install crowdstrike agent if it's not working correctly and do clean install

any suggestions to make this work are appreciated.

Thx!

**Comment by bk-cs** [comment]
When using the CrowdStrike APIs, you have to [request an access token](https://github.com/CrowdStrike/psfalcon/wiki/Request-FalconToken) before you can run any commands. For `Get-FalconUninstallToken`, you'll need `Sensor Update Policies: Write` permission.

`Get-FalconUninstallToken` expects a `device_id` value in the `Id` parameter. See examples [here](https://github.com/CrowdStrike/psfalcon/wiki/Get-FalconUninstallToken).

**Reply by dgendera** [reply]
That I understand, and as I said already I'm able to make this work and I do have those tokens so I'm able to get the maintenance token for the device and I'm able to uninstall the sensor.    What I'm trying to achieve is to automate this where I can simple script on the actual device that makes the REST API call to retrieve token and uninstall the sensor and in later stage we will re-install it using our Software Distribution system.  I'm looking for a way where I can parse as part of the command line the ClientID/ClientSecret which is required to run this cmdLet   Hope this clarifies.

**Reply by bk-cs** [reply]
https://github.com/CrowdStrike/psfalcon/wiki/Authentication#request-authorization-token-and-run-a-command
